cpp/ql/test/library-tests/ir/ir/raw_ir.expected

```diff
@@ -2602,63 +2602,59 @@ ir.cpp:
 #  571|     mu0_1(unknown)        = AliasedDefinition        : 
 #  571|     mu0_2(unknown)        = UnmodeledDefinition      : 
 #  572|     r0_3(glval<char[32]>) = VariableAddress[a_pad]   : 
-#  572|     r0_4(glval<char[1]>)  = StringConstant[""]       : 
-#  572|     r0_5(char[1])         = Load                     : &:r0_4, ~mu0_2
-#  572|     mu0_6(char[1])        = Store                    : &:r0_3, r0_5
-#  572|     r0_7(unknown[31])     = Constant[0]              : 
-#  572|     r0_8(int)             = Constant[1]              : 
-#  572|     r0_9(glval<char>)     = PointerAdd[1]            : r0_3, r0_8
-#  572|     mu0_10(unknown[31])   = Store                    : &:r0_9, r0_7
-#  573|     r0_11(glval<char[4]>) = VariableAddress[a_nopad] : 
-#  573|     r0_12(glval<char[4]>) = StringConstant["foo"]    : 
-#  573|     r0_13(char[4])        = Load                     : &:r0_12, ~mu0_2
-#  573|     mu0_14(char[4])       = Store                    : &:r0_11, r0_13
-#  574|     r0_15(glval<char[5]>) = VariableAddress[a_infer] : 
-#  574|     r0_16(glval<char[5]>) = StringConstant["blah"]   : 
-#  574|     r0_17(char[5])        = Load                     : &:r0_16, ~mu0_2
-#  574|     mu0_18(char[5])       = Store                    : &:r0_15, r0_17
-#  575|     r0_19(glval<char[2]>) = VariableAddress[b]       : 
-#  575|     mu0_20(char[2])       = Uninitialized[b]         : &:r0_19
-#  576|     r0_21(glval<char[2]>) = VariableAddress[c]       : 
-#  576|     mu0_22(char[2])       = Uninitialized[c]         : &:r0_21
-#  576|     r0_23(int)            = Constant[0]              : 
-#  576|     r0_24(glval<char>)    = PointerAdd[1]            : r0_21, r0_23
-#  576|     r0_25(unknown[2])     = Constant[0]              : 
-#  576|     mu0_26(unknown[2])    = Store                    : &:r0_24, r0_25
-#  577|     r0_27(glval<char[2]>) = VariableAddress[d]       : 
-#  577|     mu0_28(char[2])       = Uninitialized[d]         : &:r0_27
-#  577|     r0_29(int)            = Constant[0]              : 
-#  577|     r0_30(glval<char>)    = PointerAdd[1]            : r0_27, r0_29
+#  572|     r0_4(glval<char[32]>) = StringConstant[""]       : 
+#  572|     r0_5(char[32])        = Load                     : &:r0_4, ~mu0_2
+#  572|     mu0_6(char[32])       = Store                    : &:r0_3, r0_5
```

Comment:
I think what's going on here is that the extractor now gives the `StringConstant` the size that it is padded to rather than the length of the string plus the null terminator. The IR generation code would previously do the padding as a separate write starting after the null terminator (the following four instructions in the parent commit).

It seems reasonable to give this size, but I wouldn't be surprised if we rely on the old behavior in other libraries or queries.

Also, I'm not sure whether we should preserve the current behavior in the IR